File: Medium/1557. Minimum Number of Vertices to Reach All Nodes/Solution.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        List<Integer> ret = new ArrayList<>();
        int[] inDegree = new int[n];

        for (List<Integer> edge : edges)
            inDegree[edge.get(1)]++;

        for(int node = 0; node < n; node++)
            if(inDegree[node] == 0) ret.add(node);

        return ret;
    }
}

File: Medium/1557. Minimum Number of Vertices to Reach All Nodes/Solution.js
var findSmallestSetOfVertices = function(n, edges) {
    let inDegree = new Array(n).fill(0);
    let ret = [];

    for(let [u, v] of edges)
        inDegree[v]++;

    for(let node = 0; node < n; node++)
        if(!inDegree[node])
            ret.push(node);

    return ret;
};